pymeasure/instruments/tdk/tdk_base.py

```diff
@@ -0,0 +1,387 @@
+
+    address = Instrument.setting(
+        "ADR %d",
+        """Set the address of the power supply.
+
+        Valid values are integers between 0 - 30 (inclusive).""",
+        validator=strict_discrete_set,
+        values=range(0, 31)
+    )
+
+    remote = Instrument.control(
+        "RMT?", "RMT %s",
+        """Get the current remote operation of the power supply.
+
+        Valid values are ``'LOC'`` for local mode, ``'REM'`` for remote mode,
+        and ``'LLO'`` for local lockout mode.
+        """,
+        validator=strict_discrete_set,
+        values=["LOC", "REM", "LLO"]
+    )
+
+    multidrop_capability = Instrument.measurement(
+        "MDAV?",
+        """Get the multi-drop option is available on the power supply.
+
+        If return value is 0, the option is not available, if 1 it is available.
+
+        Property is UNTESTED.
+        """
+    )
+
+    master_slave_setting = Instrument.measurement(
+        "MS?",
+        """Get the master and slave settings.
+
+        Possible master return values are 1, 2, 3, and 4. The slave value is 0.
+
+        Property is UNTESTED.
+        """
+    )
+
+    repeat = Instrument.measurement(
+        "\\",
+        """Measure the last command again.
+
+        Returns output of the last command.
+        """
+    )
+
+    id = Instrument.measurement(
+        "IDN?",
+        """Get the identity of the instrument.
+
+        Returns a list of instrument manufacturer and model in the format: ``["LAMBDA", "GENX-Y"]``
+        """
+    )
+
+    version = Instrument.measurement(
+        "REV?",
+        """Get the software version on instrument.
+
+        Returns the software version as an ASCII string.
+        """
+    )
+
+    serial = Instrument.measurement(
+        "SN?",
+        """Get the serial number of the instrument.
+
+        Returns the serial number of of the instrument as an ASCII string.
+        """
+    )
+
+    last_test_date = Instrument.measurement(
+        "DATE?",
+        """Get the date of the last test, possibly calibration date.
+
+        Returns a string in the format: yyyy/mm/dd.
+        """
+    )
+
+    voltage = Instrument.control(
+        "PV?", "PV %g",
+        """Control the programmed (set) output voltage.""",
+        validator=lambda v, vs: strict_discrete_range(v, vs, step=0.01),
+        values=[0, 40],
+        dynamic=True
+    )
+
+    actual_voltage = Instrument.measurement(
+        "MV?",
+        """Measure the the actual output voltage."""
+    )
+
+    current = Instrument.control(
+        "PC?", "PC %g",
+        """Control the programmed (set) output current.""",
+        validator=lambda v, vs: strict_discrete_range(v, vs, step=0.01),
+        values=[0, 38],
+        dynamic=True
+    )
+
+    actual_current = Instrument.measurement(
+        "MC?",
+        """Measure the actual output current.
+
+        Returns a float with five digits of precision.
+        """
+    )
```

Comment:
We have a naming scheme, see https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/instrument.html#power-supplies

So current/voltage indicates the actual value, while current_setpoint indicates the setpoint.

Reply:
Updated those properties to setpoint in last commit.